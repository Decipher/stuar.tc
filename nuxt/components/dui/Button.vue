<template>
  <component
    :is="component.is"
    class="btn"
    :class="classes"
    role="button"
    v-bind="component.props || {}"
  >
    <slot />
  </component>
</template>

<script>
export default {
  props: {
    href: {
      type: String,
      default: undefined,
    },

    /**
     * The button theme
     *
     * @type {string}
     */
    theme: {
      type: String,
      default: 'neutral',
    },

    to: {
      type: String,
      default: undefined,
    }
  },

  computed: {
    classes: ({ theme }) => [
      theme ? `btn-${theme}` : undefined
    ],

    component: ({ href, to }) => {
      if (href) return { is: 'a', props: { href } }
      if (to) return { is: 'nuxt-link', props: { tag: 'button' }}
      return { is: 'button' }
    }
  }
}
</script>
