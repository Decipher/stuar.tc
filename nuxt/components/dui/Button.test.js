import { mount } from '@vue/test-utils'
import Button from './Button.vue'

describe('Button', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Button)
    expect(wrapper.vm).toBeTruthy()

    // Expect HTML to match snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })
})
