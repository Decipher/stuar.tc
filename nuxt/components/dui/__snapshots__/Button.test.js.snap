// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is a Vue instance 1`] = `"<button role=\\"button\\" class=\\"btn btn-neutral\\"></button>"`;
